eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1664,7 +1668,11 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       if (c.scriptPubKey.exists(_ != d.localShutdown.scriptPubKey) && !useSimpleClose) {
         handleCommandError(ClosingAlreadyInProgress(d.channelId), c)
       } else if (localShutdown_opt.nonEmpty || c.feerates.nonEmpty) {
-        val d1 = d.copy(localShutdown = localShutdown_opt.getOrElse(d.localShutdown), closingFeerates = c.feerates.orElse(d.closingFeerates))
+        val closeStatus1 = d.closeStatus match {
+          case initiator: CloseStatus.Initiator => initiator.copy(feerates_opt = c.feerates.orElse(initiator.feerates_opt))
+          case nonInitiator: CloseStatus.NonInitiator => nonInitiator.copy(feerates_opt = c.feerates.orElse(nonInitiator.feerates_opt)) // NB: this is the corner case where we can be non-initiator and have custom feerates
```

Comment:
Note how the "bump fee as non-initiator" wasn't properly managed in the 1st iteration.